calibrate-touchscreen/src/main.rs

```diff
@@ -0,0 +1,429 @@
+                            registry.bind::<wl_compositor::WlCompositor, _, _>(name, 1, qh, ());
+                        state.compositor = Some(compositor);
+                    }
+                    "wl_shm" => {
+                        let shm = registry.bind::<wl_shm::WlShm, _, _>(name, 1, qh, ());
+                        state.wl_shm = Some(shm);
+                    }
+                    "wl_seat" => {
+                        registry.bind::<wl_seat::WlSeat, _, _>(name, 1, qh, ());
+                    }
+                    "xdg_wm_base" => {
+                        let wm_base = registry.bind::<xdg_wm_base::XdgWmBase, _, _>(name, 1, qh, ());
+                        state.wm_base = Some(wm_base);
+                    }
+                    _ => {}
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Dispatch<wl_output::WlOutput, ()> for State {
+    fn event(
+        state: &mut Self,
+        output: &wl_output::WlOutput,
+        event: wl_output::Event,
+        _: &(),
+        _: &Connection,
+        qh: &QueueHandle<Self>,
+    ) {
+        use wayland_client::protocol::wl_output::Mode;
+
+        match event {
+            wl_output::Event::Mode { flags: WEnum::Value(flags), width, height, refresh } => {
+                if flags.contains(Mode::Current) {
+                    state.outputs.entry(output.clone())
+                        .and_modify(|fullscreen| *fullscreen = Some(FullscreenSurface{
+                            width: width,
+                            height: height,
+                            wl_surface: None,
+                            buffer: None,
+                            xdg_surface: None,
+                    }));
+
+                    state.draw_fullscreen_surfaces(qh);
+                }
+            }
+
+            _ => {}
+        }
+    }
+}
+
+fn draw_target(tmp: &mut File, (buf_x, buf_y): (u32, u32), (target_x, target_y): (f64, f64)) {
+    let centre_x = (buf_x as f64 * target_x) as i64;
+    let centre_y = (buf_y as f64 * target_y) as i64;
+    let target_size = (max(buf_x, buf_y)/80) as i64;
+    use std::{io::Write};
+    let mut buf = std::io::BufWriter::new(tmp);
+    for y in 0..buf_y as i64 {
+        for x in 0..buf_x as i64 {
+            let distance_squared =(x-centre_x)*(x-centre_x) + (y-centre_y)*(y-centre_y);
+            if distance_squared >= target_size*target_size {
+                let a = 0xFF;
+                let r = 0x3F;
+                let g = 0x3F*y as u32/buf_y;
+                let b = 0x3F*y as u32/buf_y;
+                buf.write_all(&[b as u8, g as u8, r as u8, a as u8]).unwrap();
+            }
+            else {
+                let intensity = 0xFF - 0x40*(distance_squared/(target_size*target_size/4));
+                let a = 0xFF;
+                buf.write_all(&[intensity as u8, intensity as u8, intensity as u8, a as u8]).unwrap();
+            }
+        }
+    }
+    buf.flush().unwrap();
```

Comment:
No need to flush the buffer if you don't have a `BufWriter` :thinking:

Reply:
That doesn't work for me. (It takes seconds for things to appear & update)

I can't, right now, find the example I "stole" this approach from, but it seemed a reasonable source